### PR TITLE
Adds parser for Docker config.

### LINKED ISSUE #101: Pull registry credentials from Docker config.
This should be used if credentials could not be found in defined `credHelpers` and Maven settings, but before inferring credential helpers.

Docker config may store credentials in 3 ways (or a combination):

```json
{
    "auths": {
        "registry": {
            "auth": "username:password in base64",
        },
        ...
    }
}
```

```json
{
    "auths": {
        "registry": {},
        ...
    },
    "credsStore": "credential helper name"
}
```

```json
{
    "credHelpers": {
        "registry": "credential helper name",
        ...
    }
}
```

Search for credentials in the above order, finishing if any is found.

### PR DESCRIPTION
**Pull request by coollog**
Part of #101.

Need to still add the portion to actually use this in `RetrieveCredentialsStep`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+## 0.1.2
+### Added
+- Use credentials from Docker config if none can be found otherwise ([]())
```
Comment (coollog) [inline CHANGELOG.md]: This should be populated with the next PR that adds this functionality to `jib-maven-plugin`

File: jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java [hunk 2]
```diff
@@ -35,5 +37,10 @@ public static Authorization withBasicCredentials(String username, String secret)
     return new Authorization("Basic", token);
   }
 
+  /** Creates an {@link Authorization} with a base64-encoded {@code username:password} string. */
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java]: do we/can we ensure this is only used for docker-config consumed credentials?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java]: This is intended to be used with any basic authorization token, which are in the form of `username:password`.

Comment (loosebazooka) [reply jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java]: oh yeah, I didn't really remember if Authorization was part of the api, but I mean are we only creating Authorizations withBasicToken when using docker-config?

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ *   <li>Using the credential helper from {@code credHelpers}, if available.
+ * </ol>
+ *
+ * @see <a
+ *     href="https://docs.docker.com/engine/reference/commandline/login/">https://docs.docker.com/engine/reference/commandline/login/</a>
+ */
+public class DockerConfigCredentialRetriever {
+
+  private static final Path DOCKER_CONFIG_FILE =
+      Paths.get(System.getProperty("user.home")).resolve(".docker").resolve("config.json");
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: is this true for windows/mac too?

File: jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java (hunk 2 above)
Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorizations.java]: Yea, the only place that uses this now is in the `DockerConfigCredentialRetriever`.

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java (hunk 3 above)
Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: Yes, on windows, the `user.home` would give `%USERPROFILE%`, whereas on the unix systems, it gives `$HOME`.

https://docs.docker.com/engine/reference/commandline/login/#privileged-user-requirement states the location of the config file.

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/json/DockerConfigTemplate.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Template for a Docker config file.
+ *
+ * <p>Example:
+ *
+ * <pre{@code
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/json/DockerConfigTemplate.java]: do you need to close this `pre` tag with a `>`? It's rendering funny here.

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/json/DockerConfigTemplate.java]: Oh yes

**Pull request merged**